src/gui/editors/PianoRoll.cpp

```diff
@@ -290,7 +290,7 @@ PianoRoll::PianoRoll() :
 			this, SLOT( updatePositionStepRecording( const lmms::TimePos& ) ) );
 
 	// update timeline when in record-accompany mode
-	connect(m_timeLine, &TimeLineWidget::positionChanged, this, &PianoRoll::updatePositionAccompany);
+	connect(&Engine::getSong()->getTimeline(Song::PlayMode::Song), SIGNAL(positionChanged(const lmms::TimePos&)), this, SLOT(updatePositionAccompany(const lmms::TimePos&)));
```

Comment:
```suggestion
	connect(&Engine::getSong()->getTimeline(Song::PlayMode::Song), &Timeline::positionChanged, this, &PianoRoll::updatePositionAccompany);
```
We're trying to avoid using the `SIGNAL` and `SLOT` macros in new code

Reply:
Actually, there may be a better approach than adding a signal to `Timeline`... Let me think this over some more

Reply:
I don't really like how `Timeline` and `TimeLineWidget` have implemented their signals and slots for changes in position. I think ideally `Timeline` should be the model and only it should emit any `positionChanged` signals, while `TimeLineWidget` should be the view and its `updatePosition` slot should not emit any signals. The way it is currently, the `TimeLineWidget` is trying to do the job of both a model and a view, and that complicates things.

Reply:
@DomClark Does that sound right to you? It looks to me like a proper solution to this bug may require rethinking how the timeline position signals/slots are implemented.

Reply:
I have now updated to PR to refactor everything in order to move the `positionChanged` signal from `TimeLineWidget` to `Timeline`.